### PR TITLE
Always return subquery

### PR DESCRIPTION
**Pull request by barakalon**
Always return subqueries, never paranthesized subqueryables.

### CODE REVIEW

File: sqlglot/optimizer/scope.py [hunk 1]
```diff
@@ -86,19 +86,32 @@ def _collect(self):
         self._derived_tables = []
         self._raw_columns = []
 
-        for node, *_ in _walk_next_scope(self.expression):
+        # We'll use this variable to pass state into the dfs generator.
+        # Whenever we set it to True, we exclude a subtree from traversal.
+        prune = False
```
Comment (barakalon) [inline sqlglot/optimizer/scope.py:91]: This is a wacky way to control the DFS traversal... but I kinda dig it.

Comment (tobymao) [reply sqlglot/optimizer/scope.py:91]: does this reval the lambda????

File: tests/fixtures/optimizer/optimizer.sql [hunk 2]
```diff
@@ -39,13 +39,13 @@ WITH "cte" AS (
     SELECT
       "x"."a" AS "a"
     FROM "x" AS "x"
-  ) AS "_q_0"
```
Comment (tobymao) [inline tests/fixtures/optimizer/optimizer.sql:42]: why did this get changed? is this correct?

**Review by tobymao — APPROVED** [review]

File: tests/fixtures/optimizer/optimizer.sql (hunk 2 above)
Comment (barakalon) [reply tests/fixtures/optimizer/optimizer.sql:42]: These statements were getting treated as derived tables, so they get qualified.

But I think they're actually subqueries, so I think this is correct.

File: sqlglot/optimizer/scope.py (hunk 1 above)
Comment (barakalon) [reply sqlglot/optimizer/scope.py:91]: Lambdas can reference variables in the scope they're defined. The values don't get copied. 💥 

So every time the lambda function gets called, it returns the current value of `prune`. And the lambda gets called on every node.

**Pull request merged**